src/pages/todo/components/TodoList.js

```diff
@@ -13,18 +14,76 @@ import { LIST_TODOS } from '../actions';
   };
 })
 class TodoList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      hidden: true,
+      form: { 
+        method: 'POST',
+        title: '' 
+      }
+    }
+  }
+
+  componentDidUpdate() {
+    this.refs.textbox.focus()
+  }
+
+  _submitForm() {
+    // Handle submit
+    this.props.dispatch({type: CREATE_TODO.REQUEST, data: this.state.form})
+    this.setState({form: {method: 'POST', title: ''}})
+    this.setState({hidden: true})
+  }
+
+  _showForm() {
+    this.setState({hidden: false})
+  }
+
+  _submitWhenEnter(event) {
+    if (event.keyCode !== 13) {
+      return;
+    }
+    event.preventDefault();
+    this._submitForm();
+  }
+
+  _handleChange(event) {
+    let data = {}
+    data['method'] = 'POST'
+    data[event.target.id] = event.target.value
+    this.setState({form: data})
+  }
 
   render() {
     const { todos } = this.props;
+    const { hidden } = this.state;
+    const style = {
+      display: hidden ? 'none' : 'block'
+    }
     return (
       <div className="todo-list">
         {todos.map((todo, index) => {
           return (
             <div key={index}>
-              <Link to={"/todos/" + todo.id}>{todo.title}</Link>
+              <TodoListItem 
+                todo={todo}
+              />
             </div>
           );
         })}
```

Comment:
I would like to make the input ui as in gmail todo. But in this merge request, it's ok just to put the add button.